scripts/dumpapp

```diff
@@ -76,9 +77,12 @@ def read_frames(sock):
         raise IOError('Unexpected header: %s' % code)
       break
 
-def read_input(sock, n, tag):
-  data = sock.recv(n)
-  if not data or len(data) != n:
+def read_input(sock, n, tag):  
+  data = b'';
+  while len(data) < n:
+    incoming_data = sock.recv(n - len(data))
```

Comment:
sock.recv should return an empty byte object (len(incoming_data) == 0) on EOF/error.  You have to check for this otherwise the client script will hang when the server terminates (for instance with `./scripts/dumpapp crash exit`)